core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -978,6 +1091,32 @@ private Set<String> expectedManifestListPaths(Iterable<Snapshot> snapshots, Long
         .collect(Collectors.toSet());
   }
 
+  private void createTableAndAddPartitionOnNestedFiled() throws IOException {
```

Comment:
Great to add all these tests.  I think it may fit better in "TestMetadataTablesWithPartitionEvolution" because if i understand its a partition evolution case?  But maybe its too much work to move the new tests there.

Reply:
OK, let me look it.

Reply:
`TestMetadataTablesWithPartitionEvolution` seems to be a Spark test. I would actually prefer to have the tests in `iceberg-core` (maybe in a subclass of `TestReadProjection`?)

Reply:
Ah didnt realize, yea its fine then for me where it is.

Reply:
Split into a dedicated class.